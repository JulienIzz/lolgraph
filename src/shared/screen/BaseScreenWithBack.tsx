import styled from "@emotion/native";
import { Header } from "./Header";
import { QueryBoundaries } from "../utils/QueryBoundaries";

interface ScreenProps {
  children: React.ReactNode;
  title: string;
}

export const BaseScreenWithBack = ({ children, title }: ScreenProps) => {
  return (
    <Container>
      <Header backButton title={title} />
      <QueryBoundaries>{children}</QueryBoundaries>
    </Container>
  );
};

const Container = styled.View({
  flex: 1,
});
