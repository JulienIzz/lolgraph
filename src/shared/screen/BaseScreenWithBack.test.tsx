import { render, screen } from "@testing-library/react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { View } from "react-native";
import { BaseScreenWithBack } from "./BaseScreenWithBack";

jest.mock(
  "react-native-safe-area-context",
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  () => require("react-native-safe-area-context/jest/mock").default,
);

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => ({ navigate: jest.fn() }),
}));

test("home screen with title is correctly rendered", async () => {
  const expectedHeaderTextForTest = "ScreenTitle";

  render(
    <SafeAreaProvider>
      <BaseScreenWithBack title={expectedHeaderTextForTest}>
        <View />
      </BaseScreenWithBack>
    </SafeAreaProvider>,
  );

  const HeaderOutput = screen.getByText(expectedHeaderTextForTest);

  expect(HeaderOutput).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
