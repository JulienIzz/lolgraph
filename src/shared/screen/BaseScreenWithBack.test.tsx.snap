// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen with title is correctly rendered 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      [
        {
          "backgroundColor": "#00B4D8",
          "height": 60,
          "paddingTop": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 30,
            "left": 15,
            "opacity": 1,
            "position": "absolute",
            "width": 30,
          }
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "flex": 1,
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
      </View>
      <Text
        style={
          [
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        ScreenTitle
      </Text>
    </View>
  </View>
  <View />
</View>
`;
